addons/sourcemod/scripting/freak_fortress_2.sp

```diff
@@ -6543,182 +6553,243 @@ stock GetBossIndex(client)
 	return -1;
 }
 
-stock CalcBossHealthMax(client)
+stock Calculate(client, const String:key[], const String:defaultFormula[], malformDefault)
 {
 	decl String:formula[1024], String:buffer[2];
-	new String:value[1024];
-	new bool:mustClose, bool:usePlayers, bool:canAdd, bool:valueReady;
-	new parentheses;
-	new Float:sum[128];
-	new _operator[128];
-
-	KvRewind(BossKV[Special[client]]);
-	KvGetString(BossKV[Special[client]], "health_formula", formula, sizeof(formula), "(((760.8+n)*n-1)^1.0341)+2046");
+	new String:value[1024], Float:sum[128], calcOperators:_operator[128];
+	new bool:closeBracket, bool:usePlayers, bool:useLives, bool:foundOperator, bracket;
+	new Handle:bKv = BossKV[Special[client]];
+	//This version of calculator will always deal with sum
+	KvRewind(bKv);
+	KvGetString(bKv, key, formula, sizeof(formula), defaultFormula); //"(((760.8+n)*n-1)^1.0341)+2046" <--health
 	ReplaceString(formula, sizeof(formula), " ", "");  //Get rid of spaces
 	new length=strlen(formula);
 	for(new i; i<=length; i++)
 	{
 		strcopy(buffer, sizeof(buffer), formula[i]);
-		switch(buffer[0])
+		switch (buffer[0])
 		{
 			case '(':
 			{
-				parentheses++;
-				sum[parentheses]=0.0;
-				_operator[parentheses]=0;
+				//Were in a bracket.
+				if (bracket + 1 == sizeof(_operator))
+				{
+					decl String:bossName[32];
+					KvRewind(bKv);
+					KvGetString(bKv, "name", bossName, sizeof(bossName));
+					LogError("[FF2] %s has too many operators in operation! Returning malformDefault", bossName);
+					return malformDefault;
+				}
+				closeBracket = false;
+				bracket++;
+				_operator[bracket] = Operation_None;
+				sum[bracket] = 0.0;
+				break;
 			}
 			case ')':
 			{
-				valueReady=true;
-				mustClose=true;
-			}
-			case '\0':
-			{
-				valueReady=true;
-			}
-			case 'n', 'x':
-			{
-				usePlayers=true;
-			}
-			case '+':
-			{
-				_operator[parentheses]=1;
-				canAdd=true;
-			}
-			case '-':
-			{
-				_operator[parentheses]=2;
-				canAdd=true;
+				//Close bracket now!
+				closeBracket = true;
+				break;
 			}
-			case '*':
+			default:
 			{
-				_operator[parentheses]=3;
-				canAdd=true;
+				StrCat(value, sizeof(value), buffer);
+				break;
 			}
-			case '/':
+		}
+		if (i == length && !closeBracket && !foundOperator) //Handle constant operation(eg: "health_formula" 5000
```

Comment:
If you could find a non-hacky way to figure out if we have a constant value that would be great :P.